### PR TITLE
server: Fix io flow stat for snapshot sending

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Ref #15990 

What's Changed:

```commit-message
Fix io flow stat for snapshot sending. Its io flow is not taken into account previously 
because the io flow is recorded only when using `file_system::File` which is a wrapper of `std::fs::File`.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: src/server/snap.rs [hunk 1]
```diff
@@ -128,6 +129,7 @@ impl Stream for SnapChunk {
             n if n > SNAP_CHUNK_LEN => vec![0; SNAP_CHUNK_LEN],
             n => vec![0; n],
         };
+        let _with_io_type = WithIoType::new(self.io_type);
```
Comment (LykxSassinator) [inline src/server/snap.rs:132]:
```suggestion
        let _guard = WithIoType::new(self.io_type);
```
Better to use `_guard`.

**Review by LykxSassinator — COMMENTED** [review]
rest LGTM

File: src/server/snap.rs (hunk 1 above)
Comment (Connor1996) [reply src/server/snap.rs:132]: Other places use this naming, so let's keep it

**Review by hbisheng — APPROVED** [review]
LGTM

**Review by glorv — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**